Dockerfile

```diff
@@ -1,22 +1,34 @@
-# To build:
-#    make docker-build
-# To push:
-#    make docker-push
-
-FROM golang:1.22.4-bullseye as build
+# Step 1: Build Go API
+FROM golang:1.22.4-bullseye as api-build
 ARG GIT_COMMIT
 
 WORKDIR /src/wallet-backend
-ADD go.mod go.sum ./
+COPY go.mod go.sum ./
 RUN go mod download
-ADD . ./
+COPY . ./
 RUN go build -o /bin/wallet-backend -ldflags "-X main.GitCommit=$GIT_COMMIT" .
 
 
-FROM ubuntu:22.04
+# Step 2: Install Stellar Core and copy over app binary
+FROM ubuntu:jammy AS core-build
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends ca-certificates curl wget gnupg apt-utils gpg && \
+    curl -sSL https://apt.stellar.org/SDF.asc | gpg --dearmor >/etc/apt/trusted.gpg.d/SDF.gpg && \
+    echo "deb https://apt.stellar.org jammy stable" >/etc/apt/sources.list.d/SDF.list && \
+    echo "deb https://apt.stellar.org jammy testing" >/etc/apt/sources.list.d/SDF-testing.list && \
+    echo "deb https://apt.stellar.org jammy unstable" >/etc/apt/sources.list.d/SDF-unstable.list && \
+    apt-get update && \
+    apt-get install -y stellar-core=21.0.0-1872.c6f474133.jammy && \
```

Comment:
It would be worth parameterising the version so that you don't have to edit the Dockerfile each time new core is released.

Reply:
Good point @jacekn! To be honest I had it parameterized before, but what happened is that that way I had to update two different places when upgrading the version: the Makefile's default value and the docker-compose.yaml so that they're matching. So I ended up changing it to be fixed in a central place (the Dockerfile).

In Vibrant at least, every time we wanted to upgrade the version I made a deploy changing the value in the Makefile. I guess we could also update the Jenkins pipeline. But I'm wondering how this is usually being done in Horizon / Soroban RPC @jacekn?

Reply:
In case of horizon we choose core version from a dropdown in Jenkins and pass as parameter:
https://github.com/stellar/go/blob/master/services/horizon/docker/Makefile

But if you find it easier to maintain it here that's OK. Another option could be to make it a parameter but have default version set in the Makefile. That way anybody can build with any version and you'd open PRs against the Makefile to update the default version.

Reply:
I've changed it back here: [45cb905](https://github.com/stellar/wallet-backend/pull/30/commits/45cb905de44d5e47121842c851cae0780878619e)
I think it will be easier for other dev teams to configure it all in one place (the Makefile args).

Reply:
Good to know about how Horizon is doing it. I think in our case for now we don't need it to be configurable to that level (with the Jenkins dropdown) but will be easy to add it if necessary.